Src/Application/Services/TodoItems/ItemQueries/IItemsQueryService.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Todo.Services.TodoItems.Queries.GetItem;
+using Todo.Services.TodoItems.Queries.Lookups.ChildItems;
+using Todo.Services.TodoItems.Queries.Lookups.ParentItems;
+
+namespace Todo.Application.Services.TodoItems.ItemQueries
+{
+    public interface IItemsQueryService : IGetItemService,
+                                          IChildItemsLookupService,
+                                          IParentItemsLookupService
+    {
+    }
```

Comment:
Definitely makes sense, however, this interface is used within the decorator implementation... so I'm assuming it would be better to remove this interface and convert the `ItemsQueryService.cs` class into a composition of the query services, however, create cache decorator services for each query service? 🤔

Reply:
Actually! I just realised that is exactly what I should be doing... So creating cached services per query/command service is better, as it satisfies the SRP and OCP, and not only that, if there was a particular query in which I didn't want to cache, I could so easily as I wouldn't "decorate" that query service! Also, caching is more of an infrastructure layer concern, NOT a Core layer concern! Thus, all caching implementations/decorators should be in the infrastructure layer \o/